api/app/src/main/kotlin/packit/service/RoleService.kt

```diff
@@ -52,17 +55,37 @@ class BaseRoleService(
         saveRole(createRole.name, permissions)
     }
 
-    @Transactional
     override fun deleteRole(roleName: String)
     {
-
         if (!roleRepository.existsByName(roleName))
         {
             throw PackitException("roleNotFound", HttpStatus.BAD_REQUEST)
         }
         roleRepository.deleteByName(roleName)
     }
 
+    override fun addPermissionsToRole(roleName: String, addRolePermissions: List<UpdateRolePermission>)
+    {
+        val role = roleRepository.findByName(roleName)
+            ?: throw PackitException("roleNotFound", HttpStatus.BAD_REQUEST)
+
+        val rolePermissionsToAdd = rolePermissionService.getRolePermissionsToUpdate(role, addRolePermissions)
+        if (rolePermissionsToAdd.any { role.rolePermissions.contains(it) })
+        {
+            throw PackitException("rolePermissionAlreadyExists", HttpStatus.BAD_REQUEST)
+        }
+
+        role.rolePermissions.addAll(rolePermissionsToAdd)
+        roleRepository.save(role)
+    }
+
+    override fun removePermissionsFromRole(roleName: String, removeRolePermissions: List<UpdateRolePermission>)
+    {
+        val role = roleRepository.findByName(roleName)
+            ?: throw PackitException("roleNotFound", HttpStatus.BAD_REQUEST)
+        rolePermissionService.removeRolePermissionsFromRole(role, removeRolePermissions)
```

Comment:
It seems a bit odd to check that permissions being added don't yet exist here in `RoleService` but that permissions deing deleted do exist, in `RolePermissionService` service. Should both be in the same service, probably `RolePermissionsService`